### PR TITLE
feat: lock server patch

### PR DESCRIPTION
**Pull request by kindermoumoute**

### CODE REVIEW

File: api/instance/v1/server_utils.go [hunk 1]
```diff
@@ -13,6 +13,14 @@ import (
 	"github.com/scaleway/scaleway-sdk-go/utils"
 )
 
+type UpdateServerRequest updateServerRequest
+
+// updateServer update server
+func (s *API) UpdateServer(req *UpdateServerRequest, opts ...scw.RequestOption) (*UpdateServerResponse, error) {
+	defer lockServer(req.Zone, req.ServerID).Unlock()
+	return s.updateServer((*updateServerRequest)(req), opts...)
```
Comment (DenBeke) [inline api/instance/v1/server_utils.go]: Maybe I'm overlooking something, but where do you lock the server resource?

Comment (kindermoumoute) [reply api/instance/v1/server_utils.go]: in `lockServer` 👍

Comment (DenBeke) [reply api/instance/v1/server_utils.go]: But `lockServer` is executed in the defer?

Comment (kindermoumoute) [reply api/instance/v1/server_utils.go]: Onyl the last function is deferred

Comment (jerome-quere) [reply api/instance/v1/server_utils.go]: Unlock is defered not lock

Comment (DenBeke) [reply api/instance/v1/server_utils.go]: TIL...

**Pull request merged**